src/modules/dlang.rs

```diff
@@ -0,0 +1,112 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::dlang::DLangConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current D version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("dlang");
+    let config = DLangConfig::try_load(module.config);
+    let is_dlang_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .is_match();
+
+    if !is_dlang_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => context
+                    .exec_cmd("dmd", &["--version"])
+                    .and_then(|output| parse_dlang_version(output.stdout.trim()))
+                    .map(Ok),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `dlang`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn parse_dlang_version(dlang_version: &str) -> Option<String> {
+    let version = dlang_version
+        .split('\n')
+        .next()?
+        .split_whitespace()
+        .nth(3)?;
+
+    Some(version.to_owned())
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Please add tests for dub manifests: `dub.json` and `dub.sdl`
https://dub.pm/package-format-json.html
https://dub.pm/package-format-sdl.html